src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/support/SupportService.kt

```diff
@@ -6,21 +6,25 @@ import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.HdcService
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.IS91DeterminationService
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.prison.PrisonerSearchApiClient
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.prison.PrisonerSearchPrisoner
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.DeliusApiClient
 
 @Service
 class SupportService(
   private val prisonerSearchApiClient: PrisonerSearchApiClient,
   private val hdcService: HdcService,
   private val eligibilityService: EligibilityService,
   private val iS91DeterminationService: IS91DeterminationService,
+  private val deliusApiClient: DeliusApiClient,
 ) {
   fun getIneligibilityReasons(prisonNumber: String): List<String> {
     val prisoners = prisonerSearchApiClient.searchPrisonersByNomisIds(listOf(prisonNumber))
     if (prisoners.size != 1) {
       error("Found ${prisoners.size} prisoners for: $prisonNumber")
     }
     val prisoner = prisoners.first()
-    val reasons = eligibilityService.getIneligibilityReasons(prisoner)
+    val deliusRecord = deliusApiClient.getOffenderManagers(listOf(prisoner.prisonerNumber)).first()
+    val eligibility = eligibilityService.getEligibilityAssessments(listOf(prisoner), mapOf(prisoner.prisonerNumber to deliusRecord.team.provider.code))[prisoner.prisonerNumber]!!
```

Comment:
can use `first` here...

Reply:
extracting out `mapOf(prisoner.prisonerNumber to deliusRecord.team.provider.code)` might be nice too